aiohttp/http_writer.py

```diff
@@ -16,7 +17,12 @@
 
 class StreamWriter(AbstractStreamWriter):
 
-    def __init__(self, protocol, transport, loop):
+    def __init__(self, protocol, transport, loop, on_chunk_sent=None):
+        assert (
```

Comment:
The assert is in the hot-pass route.
I understand your striving for type checking but all `StreamWriter` usage is under the library control, we always should pass the proper values for `on_chunk_sent` parameter.
Please remove the assert statement.

Reply:
done